src/main/java/org/jabref/gui/importer/ImportAction.java

```diff
@@ -135,55 +130,14 @@ private ParserResult mergeImportResults(List<ImportFormatReader.UnknownFormatImp
                 continue;
             }
             ParserResult parserResult = importResult.parserResult;
-            List<BibEntry> entries = parserResult.getDatabase().getEntries();
-            resultDatabase.insertEntries(entries);
+            new BibDatabaseMerger().merge(resultDatabase, parserResult.getDatabase());
```

Comment:
The merge method below checks for duplicates etc, but this is already done in the import dialog where the user can also decide to overwrite the duplication check. This is behavior that we should keep, i.e. here we really need to add the entries from one db into the other one.